src/core/utils/modal.tsx

```diff
@@ -40,6 +41,16 @@ function Modal({
   const dispatch = useDispatch();
   const { modalIds } = useSelector((s: ModalIdsProps) => s.modal);
 
+  const close = useCallback(() => {
+    // This is to ensure that it only closes the top modal (multiple modals can be open at once)
+    const index = modalIds.indexOf(modalId);
+    if (index >= 0 && modalIds[index] === modalId) {
```

Comment:
Please reconsider this code. 

I appreciate the intention in the comment but I do not understand the implementation. If `const index = modalIds.indexOf(modalId)` then I cannot see when `modalIds[index] === modalId` would not be true. If that is the case then it is only a test for `index >= 0`. I had expected a test for whether the modal id was the last entry in the array.